### ISSUE TITLE
Can't use some imports from libc crate

### ISSUE THREAD

**Issue by thomcc** [intellij-rust/intellij-rust#2532]
Initially this seemed like the same bug as #1622, but I think it's different.

Errors are reported when using the imported items, not during the import. Also, it only happens for some imported items and it depends on how you import them. Weird, I know.

```rust
extern crate libc;

// Busted.
use libc::c_char;
// Also busted.
use libc::c_long;


// Works if the import is aliased, but `use libc::c_ulong` is busted.
use libc::{c_ulong as my_c_ulong};

type c_short = libc::c_short;

// Also works (non-target dependent types don't seem effected)
use libc::int8_t;

fn main() {
    let cc1: c_char = b'x';

    // Fully qualified name works
    let cc2: libc::c_char = b'y';

    let cl: c_long = 100;

    let cs: c_short = 100;

    // It works if you give it an alias
    let cc2: my_c_ulong = 300;

    let byte: int8_t = 30;

}
```

Screenshot showing the erroring portions:

<img width="954" alt="screen shot 2018-05-15 at 12 53 58 pm" src="https://user-images.githubusercontent.com/860665/40080196-1d1d5684-583f-11e8-8744-a51b94fce6e3.png">

My Cargo.toml is just

```toml
[package]
# blah blah blah...

[dependencies]
libc = "*"
```

**Comment by dvic** [comment]
FYI, I'm see the same problem.

**Comment by aldanor** [comment]
It's gotten worse in the recent updates... Probably because of their use of `cfg_if! {}` macro for compile-time logic and intellij-rust's inability to parse it properly.

**Comment by thomcc** [comment]
If you can, it's worth using the exports from std::os::raw instead. doesn't work if you actually need functions from libc though...

**Comment by mili-l** [comment]
see https://github.com/intellij-rust/intellij-rust/pull/9729
